tests/unit/conftest.py

```diff
@@ -189,11 +191,29 @@ def acc_client():
 
 
 class MockPathLookup(PathLookup):
-    def __init__(self, cwd='source_code/samples', sys_paths: list[Path] | None = None):
-        super().__init__(Path(__file__).parent / cwd, sys_paths or [])
-
-    def change_directory(self, new_working_directory: Path) -> 'MockPathLookup':
-        return MockPathLookup(new_working_directory, self._sys_paths)
+    """A path look up for the testing code samples."""
+
+    def __init__(
+        self,
+        cwd=Path(__file__).parent / "source_code/samples",
+        sys_paths: list[Path] | None = None,
+    ):
+        super().__init__(cwd, sys_paths or [])
+
+        self.successfully_resolved_paths = set[Path]()  # The paths that were successfully resolved
```

Comment:
Note that this attribute should be used in tests instead of passing a `set` for the linted paths as this is correct and it actually was a bug that the paths were not resolved